packages/fx-core/src/plugins/resource/bot/dotnet/constants.ts

```diff
@@ -17,4 +17,10 @@ export class PathInfo {
 export class RegularExpr {
   static readonly botId = /\$botId\$/g;
   static readonly botPassword = /\$bot-password\$/g;
+  static readonly clientId = /\$clientId\$/g;
```

Comment:
There is no need to use regularExp here, I think simple string replacement is work fine and it's faster too

Reply:
Updated, thx!